src/main/java/org/gridsuite/study/server/repository/StudyRepository.java

```diff
@@ -1,117 +1,46 @@
-                        publicStudyRepository.deleteByStudyNameAndUserId(studyName, userId),
-                        publicAndPrivateStudyRepository.deleteByStudyNameAndUserId(studyName, userId)).then();
-    }
+    List<StudyEntity> findAllByUserId(String userId);
 
-    public Mono<Void> updateLoadFlowState(String studyName, String userId, LoadFlowStatus lfStatus) {
-        return Mono.zip(publicAndPrivateStudyRepository.updateLoadFlowState(studyName, userId, lfStatus),
-                        privateStudyRepository.updateLoadFlowState(studyName, userId, lfStatus),
-                        publicStudyRepository.updateLoadFlowState(studyName, userId, lfStatus))
-                .then();
-    }
+    Optional<StudyEntity> findByUserIdAndStudyName(String userId, String name);
 
-    public Mono<Void> updateLoadFlowResult(String studyName, String userId, LoadFlowResultEntity loadFlowResult) {
-        return Mono.zip(privateStudyRepository.updateLoadFlowResult(studyName, userId, loadFlowResult),
-                        publicStudyRepository.updateLoadFlowResult(studyName, userId, loadFlowResult),
-                        publicAndPrivateStudyRepository.updateLoadFlowResult(studyName, userId, loadFlowResult))
-                .then();
-    }
+    void deleteByUserIdAndStudyName(String userId, String name);
 
-    public Mono<Void> updateSecurityAnalysisResultUuid(String studyName, String userId, UUID securityAnalysisResultUuid) {
-        return Mono.zip(privateStudyRepository.updateSecurityAnalysisResultUuid(studyName, userId, securityAnalysisResultUuid),
-                        publicStudyRepository.updateSecurityAnalysisResultUuid(studyName, userId, securityAnalysisResultUuid),
-                        publicAndPrivateStudyRepository.updateSecurityAnalysisResultUuid(studyName, userId, securityAnalysisResultUuid))
-                .then();
-    }
+    @Query(value = "SELECT * from study WHERE userId = :userId or isPrivate = false", nativeQuery = true)
+    List<StudyEntity> getStudyList(@Param("userId") String userId);
 
-    public Mono<Void> updateLoadFlowParameters(String studyName, String userId, LoadFlowParametersEntity parameters) {
-        return Mono.zip(publicAndPrivateStudyRepository.updateLoadFlowParameters(studyName, userId, parameters),
-                        publicStudyRepository.updateLoadFlowParameters(studyName, userId, parameters),
-                        privateStudyRepository.updateLoadFlowParameters(studyName, userId, parameters)
-        ).then();
-    }
+    @Transactional
+    @Modifying(flushAutomatically = true, clearAutomatically = true)
+    @Query(value = "UPDATE study SET loadFlowStatus = :#{#loadFlowStatus.name()} WHERE userId = :userId and studyName = :studyName", nativeQuery = true)
+    int updateLoadFlowStatus(@Param("studyName") String studyName, @Param("userId") String userId, @Param("loadFlowStatus") LoadFlowStatus loadFlowStatus);
```

Comment:
why return int here and void below ?

Reply:
why transactionnal ?

Reply:
we need @Transactional when modifying data even with @Modifying that is only an annotation to let Spring Data know you have a @Query that changes stuff